metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -175,7 +180,14 @@ class MetalsLanguageServer(
   val parseTrees = new BatchedFunction[AbsolutePath, Unit](paths =>
     CancelableFuture(
       buildServerPromise.future
-        .flatMap(_ => Future.sequence(paths.distinct.map(compilers.didChange)))
+        .flatMap(_ =>
+          Future {
```

Comment:
minor, but since you're not using Future.sequence anymore this can just be map, without creating a new Future here

Reply:
Change it a bit to fix some issues.